Make logging easier to read by prettyprinting the json

KorpelainenTommi opened the pull request:
Format json output to be easier to read.

From this:
![image](https://user-images.githubusercontent.com/105944267/212065558-8377d201-6231-4015-a408-ef661737b1a7.png)

To this:
![image](https://user-images.githubusercontent.com/105944267/212065617-787dae91-fa63-4f7a-93ba-0b185760856f.png)

Commit:
Make logging easier to read by prettyprinting the json

jaakkoha-bytecraft commented:
Logging frameworks expect that one line corresponds to one log event -- which means that this should only be done for development.

At least AWS handles having `\r` as the line separator as one wants, meaning that it won't break the log event but will be rendered as a newline  in the UI; other systems _might_ work similarily. I guess replacing `\n\r?` with `\r` for production would do the trick?

jniemii commented:
As Jaakko said, production should always have one newline per logline - but pretty printing could be enabled for development mode. Maybe modify the PR to conditionally enable pretty printing based on NODE_ENV (see level() in logger.ts)?

samikuikka committed:
real hovering on the cypress tests

samikuikka committed:
Merge branch 'main' of https://github.com/bytecraftoy/aalto-2022 into main

samikuikka committed:
Merge branch 'main' of https://github.com/bytecraftoy/aalto-2022 into main

samikuikka committed:
Merge branch 'main' of https://github.com/bytecraftoy/aalto-2022 into main

samikuikka committed:
Merge branch 'main' of https://github.com/bytecraftoy/aalto-2022 into main

samikuikka committed:
Merge branch 'main' of https://github.com/bytecraftoy/aalto-2022 into main

samikuikka committed:
fix for frontend

Commit:
Add cross-env to fix issue with environment variables on Windows

Commit:
Merge branch 'main' of github.com:bytecraftoy/aalto-2022 into pretty_print_structured_logging
Merge main

Commit:
Added conditional prettyprinting

Commit:
Merge branch 'add_cross_env' of github.com:bytecraftoy/aalto-2022 into pretty_print_structured_logging
Merge add_cross_env

KorpelainenTommi committed:
Io box buttons 2 (#86)

* login and register buttons to the navigation bar

* start of the login page, currently without validation and not coonnected to backend

* fixing input element

* moved cutominput and textarea to same folder

* Reusing prop information in textarea and custominput

* custom hoom for input and textarea for showing errors

* deleted unecessary import from textarea

* moved the PromptIOBox to ContentPanelPrompts component because it's not reusable in whole application

* custom hook for input elements + taking out the margin from filled button to fix bugs

* loggin error message v1

* closing the notification after a while from opening it

* small ui changes

* New IOBoxButtons

* Remove unrelated change, probably make this another pull request later if not lazy

* Remove errors from IOBoxBar since it's unused

* Prettier

* Prettier again, merge more stuff

* Re-add changes that were lost when merging from main

* Remove errors as it is redundant for IOBoxButtonm

* Added responsive design (#88)

* Added responsive design

* Merge bulk add (#89)

* Basic bulk adding functionality

Basic functionality for adding multiple boxes at once using pre-set buttons for adding 5 and 10 boxes at once.

* Dropdown menu

Added basic dropdown menu for secondary panel functions. Currently can add 5 or 10 I/O boxes.

* prettier on last commit

Prettier ran on previous commit

* Styling dropdown menu and cleaning the code

Changed the dropdown menu styling to utilize the color pallet and cleaned some redundant code.

* Styling updated

Further updated styling and removed the need for colorPalette in DropdownMenu

Co-authored-by: PyryManninen <pyry.manninen@gmail.com>
Co-authored-by: Tommi Korpelainen <tommi.korpelainen@aalto.fi>

Co-authored-by: Tommi Korpelainen <tommi.korpelainen@aalto.fi>
Co-authored-by: PyryManninen <pyry.manninen@gmail.com>

Co-authored-by: Sami Kuikka <sami.kuikka@aalto.fi>
Co-authored-by: Tommi Korpelainen <tommi.korpelainen@aalto.fi>
Co-authored-by: PyryManninen <pyry.manninen@gmail.com>
